editoast/src/views/timetable.rs

```diff
@@ -338,14 +338,22 @@ async fn conflicts(
         );
     }
     let conflict_detection_request = ConflictDetectionRequest {
+        infra: infra_id,
+        expected_version: infra.version,
         trains_requirements,
-        infra_id,
+        work_schedules: None,
     };
 
     // 3. Call core
-    let conflicts = conflict_detection_request.fetch(&core_client).await?;
+    let conflict_detection_response = conflict_detection_request.fetch(&core_client).await?;
 
-    Ok(Json(conflicts.conflicts))
+    let conflicts = conflict_detection_response
+        .conflicts
+        .into_iter()
+        .map(|c| c.conflict)
+        .collect();
+
+    Ok(Json(conflicts))
```

Comment:
Is there a reason for transforming into an iterator to recollect just after? Can't you just do the following?

```suggestion
    Ok(Json(conflict_detection_response.conflicts))
```

Reply:
You might have missed it, but with this, I'm removing the work_schedule_ids data to not have to send an empty list to the frontend for no reason (`ConflictDetectionResponse` contains `conflicts: Vec<ConflictWithWorkSchedule>`, and I remove the `work_schedule_ids` field to only send a `Vec<Conflict>` to the frontend).